src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -2997,6 +2996,12 @@ void StorageReplicatedMergeTree::shutdown()
         std::unique_lock lock(data_parts_exchange_endpoint->rwlock);
     }
     data_parts_exchange_endpoint.reset();
+
+    /// We clear all parts after stopping all background operations. It's
+    /// important, because background operations can produce temporary parts
+    /// which will remove themselfs in their descrutors. If so, we may have race
+    /// condition between our remove call and background process.
+    clearOldPartsFromFilesystem(true);
```

Comment:
Why not to do the same for [StorageMergeTree](https://github.com/ClickHouse/ClickHouse/blob/c7d9094a7a42488a84b8ed463bd9d2655a9f1c02/src/Storages/StorageMergeTree.cpp#L126)?